src/main/java/com/amazon/ion/impl/macro/EncodingContext.kt

```diff
@@ -6,13 +6,19 @@ package com.amazon.ion.impl.macro
  * When we implement modules, this will likely need to be replaced.
  * For now, it is a placeholder for what is to come and a container for the macro table.
  */
-class EncodingContext(macroTable: Map<MacroRef, Macro>) {
+class EncodingContext {
 
-    val macroTable = macroTable.toMutableMap()
+    val macroTable: MacroTable
+    val isMutable: Boolean
+
+    @JvmOverloads
+    constructor(macroTable: MacroTable, isMutable: Boolean = true) {
+        this.macroTable = macroTable
+        this.isMutable = isMutable
+    }
 
     companion object {
-        // TODO: Replace this with a DEFAULT encoding context that includes system macros.
         @JvmStatic
-        val EMPTY = EncodingContext(emptyMap())
+        val DEFAULT = EncodingContext(SystemMacro.SYSTEM_MACRO_TABLE, false)
```

Comment:
This is obviously a nitpick, but is there a way to tell Kotlin to generate a getter spelled `getDefault()`? `getDEFAULT()` is brutal to look at.

Reply:
Thanks Matt!